### PR TITLE
improvement(client): reoganize ContainerRuntime export

### PR DESCRIPTION
**Pull request by jason-ha**
to avoid internal deprecation

Also add lint config for examples imports.

### CODE REVIEW

**Comment by markfields** [comment]
Nice, I was wondering if something like this was possible.

Since there are a handful of infra/unrelated changes at the moment, I'd rather get those merged straight into `main`, and not include them in my PR.  Obviously replacing the unnecessary deprecation allowances with that one `internal.ts` file would be cool though (But I don't mind that landing afterwards either)

File: .changeset/plain-mails-tan.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+---
+
+The ContainerRuntime class is now deprecated
+
+The class `ContainerRuntime` is deprecated and will no longer be exported starting in version 2.20.0.
+Additionally, function signatures on `BaseContainerRuntimeFactory` have been updated from using the type `ContainerRuntime` to using `IContainerRuntime`.
+
+There are two possible migration paths to stop using `ContainerRuntime`:
+
+* When using it as a type, replace it with an interface like `IContainerRuntime` (or in rare cases, `IRuntime`)
```
Comment (jzaffiro) [inline .changeset/plain-mails-tan.md:16]:
```suggestion
* When using it as a type, replace it with an interface like `IContainerRuntime` (or in rare cases, `IRuntime`).
```

File: .changeset/plain-mails-tan.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+* When using the static function `ContainerRuntime.loadRuntime` replace it with the free function `loadContainerRuntime`.
```
Comment (jzaffiro) [inline .changeset/plain-mails-tan.md:17]:
```suggestion
* When using the static function `ContainerRuntime.loadRuntime`, replace it with the free function `loadContainerRuntime`.
```

**Review by jzaffiro — APPROVED** [review]
A few nits, but docs look good to me!

**Comment by jason-ha** [comment]
Hmm. The target was set to Mark's branch. And GitHub auto-retargets once that branch merged. .. but no need since Mark already merged.
To anyone looking at this later for the technique, only the last commit ([1503238](https://github.com/microsoft/FluidFramework/commit/1503238fb1a163449ab48a78ee400520f0c3c9fc)) is of interest.
Policy would require internal export be linted which would be a bit annoying.

**Pull request closed without merging**